sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -565,16 +566,24 @@ class depends_on {
   public:
     template<typename... NodeTN>
     depends_on(NodeTN... nodes);
+
+    depends_on(const event& depEvent);
+
+    depends_on(const std::vector<event>& depEvents);
 };
 }
 ----
 
 The API for explicitly adding nodes to a `command_graph` includes a
 `property_list` parameter. This extension defines the `depends_on` property to
-be passed here. `depends_on` defines any `node` objects for the created node to
-be dependent on, and therefore form an edge with. These nodes are in addition to
-the dependent nodes identified from the command-group requisites of the created
-node.
+be passed here. `depends_on` may be used in two ways:
+
+* Passing nodes from the same `command_graph` which will create dependencies and
+graph edges between those nodes and the node being added. 
+
+* Passing SYCL events will create runtime dependencies for execution of the
+graph node but will only create edges if those events are associated with other
+nodes in the same graph.
```

Comment:
Thanks I think that is clearer.

>What happens if the event represents a node recorded into a different graph?

In that case still only a runtime dependency is created. For easier reasoning about the structure of the graph I think it's important not to call these edges, otherwise this dependency would essentially make them into a single graph.

Reply:
> Edit: Actually perhaps your question is more about what if the event passed is from the recorded queue submission, rather than obtained via `get_event`?

Yes, that was my question.

Reply:
I think right now this would have to still be an error and we should advise to use `get_event()` and `dynamic_event`s to establish these dependencies. This is more explicit which I think is good.

It might be possible to support this in the runtime but there is a very high implementation burden there I think. Neither the graph runtime or UR command-buffers are designed to accomodate that sort of sharing.